# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1310846781 25200
#      Sat Jul 16 13:06:21 2011 -0700
# Node ID 8f58f087df7f2587e15cd01ec1d47ddcc3a3fbc5
# Parent  efd924e19ff71ec9510ba2d3422bb910b7e1bdbd
codesprint: Add tests for single()

* ov-flt-re-mat.cc: Add tests for single()

diff --git a/src/ov-flt-re-mat.cc b/src/ov-flt-re-mat.cc
--- a/src/ov-flt-re-mat.cc
+++ b/src/ov-flt-re-mat.cc
@@ -834,8 +834,37 @@ Convert @var{x} to single precision type
           OCTAVE_TYPE_CONV_BODY3 (single, octave_float_matrix, octave_float_scalar);
         }
     }
   else
     print_usage ();
 
   return octave_value ();
 }
+
+/*
+
+%!assert (class (single(1)), "single")
+%!assert (class (single(1 + i)), "single")
+%!assert (class (single (int8 (1))), "single")
+%!assert (class (single (uint8 (1))), "single")
+%!assert (class (single (int16 (1))), "single")
+%!assert (class (single (uint16 (1))), "single")
+%!assert (class (single (int32 (1))), "single")
+%!assert (class (single (uint32 (1))), "single")
+%!assert (class (single (int64 (1))), "single")
+%!assert (class (single (uint64 (1))), "single")
+%!assert (class (single (true)), "single")
+%!assert (class (single ("A")), "single")
+%!error (single (sparse (1)))
+%!test
+%! x = diag ([1 3 2]);
+%! y = single (x);
+%! assert (class (x), "double");
+%! assert (class (y), "single");
+%!test
+%! x = diag ([i 3 2]);
+%! y = single (x);
+%! assert (class (x), "double");
+%! assert (class (y), "single");
+
+*/
+
